File: 수학(구현)/백준_13015_별찍기23/별찍기23.cpp
#include <iostream>
#include <string.h>     //memset
using namespace std;

char arr[200][400];

void set(int n, int r, int idx){
    int row = 2*n-1;
    int col = 4*n-3;

    memset(arr[r], ' ', col + 1 - idx);

    if (r == 0 || r == row-1) {
        memset(arr[r], '*', n);
        memset(arr[r] + col - n, '*', n);
    }
    else {
        memset(arr[r] + idx, '*', 1);
        memset(arr[r] + idx + (n - 1), '*', 1);
        memset(arr[r] + col - idx - 1, '*', 1);
        memset(arr[r] + col - idx - n, '*', 1);
    }
}

int main() {
    int n = 0;
    scanf("%d", &n);
    int idx = 0;

    for(int i = 0; i< 2*n-1; i++){
        set(n, i, idx);
        if(i < n- 1) idx++;
        else idx--;
    }

    for(int i = 0; i< 2 * n - 1; i++ ) printf("%s\n", arr[i]);
    return 0;
}

File: 수학(구현)/백준_13015_별찍기23/별찍기23.py
n = int(input())
arr = [[' ' for _ in range(400)] for _ in range(200)]
idx = 0

def set(n, r, idx):
    row = 2*n-1
    col = 4*n-3

    for i in range(0, col - idx + 1): arr[r][i] = ' '

    if r == 0 or r == row - 1:
        for i in range(0,n): arr[r][i] = '*'
        for i in range(col - n, col): arr[r][i] = '*'
    else:
        arr[r][idx] = '*'
        arr[r][idx + n - 1] = '*'
        arr[r][col - idx - 1] = '*'
        arr[r][col - idx - n] = '*'

    for i in range(0, col - idx + 1): print(f'{arr[r][i]}', end='')

for i in range(0, 2*n-1):
    set(n, i, idx)
    if i < n-1: idx+=1
    else: idx-=1
    print()
